source/common/config/utility.h

```diff
@@ -209,13 +211,50 @@ class Utility {
     return *factory;
   }
 
+  template <class Factory> static Factory& getAndCheckFactoryByType(absl::string_view type) {
+    if (type.empty()) {
+      throw EnvoyException("Provided type for static registration lookup was empty.");
+    }
+
+    Factory* factory = Registry::FactoryRegistry<Factory>::getFactoryByType(type);
+
+    if (factory == nullptr) {
+      throw EnvoyException(
+          fmt::format("Didn't find a registered implementation for type: '{}'", type));
+    }
+
+    return *factory;
+  }
+
   /**
    * Get a Factory from the registry with error checking to ensure the name and the factory are
    * valid.
    * @param message proto that contains fields 'name' and 'typed_config'.
    */
   template <class Factory, class ProtoMessage>
   static Factory& getAndCheckFactory(const ProtoMessage& message) {
+    const ProtobufWkt::Any& typed_config = message.typed_config();
+    static const std::string struct_type =
```

Comment:
It should be possible to make these `const std::string&` I think.

Reply:
Yeah, makes sense since descriptors reside permanently.